org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/extensions/contentmodel/participants/ContentModelCompletionParticipant.java

```diff
@@ -83,33 +98,360 @@ public void onTagOpen(ICompletionRequest request, ICompletionResponse response)
+					particleIndex++;
+					XSTerm term = particle.getTerm();
+					if(term.getType() == XSConstants.ELEMENT_DECLARATION) {
+						//populates list with all possible options for this declaration
+						//considers if it is a reference/substitution groups.
+						parentCMElement.getDocument().collectElement((XSElementDeclaration) term, currentElements);
+					}
+					else if(term.getType() == XSConstants.MODEL_GROUP){
+						currentElements.addAll(getNextValidDeclsFromModelGroup(element, parentCMElement, particle, prefix, forceUseOfPrefix, request, response));
+					}
+					else if(term.getType() == XSConstants.WILDCARD) {
+							
+						DOMDocument domDocument = element.getOwnerDocument();
+						DOMElement rootElement = domDocument.getDocumentElement();
+						collectElementsFromWildcard(term, currentElements, rootElement, prefix, request);
+					}
+					boolean addDecl = true;
+					for (DOMElement childDom: domElements) {
+						if(doElementDeclsContainElement(currentElements, childDom)) { // element already exists in document
+							if(modelGroupType == XSModelGroup.COMPOSITOR_CHOICE) { // choice only allows for one of it's options
+								boolean isUnderMaxOccurs = modelGroupParticle.getMaxOccurs() > element.getNumberOfChildElementsWithTagName(childDom.getTagName());
+								if(isUnderMaxOccurs || modelGroupParticle.getMaxOccursUnbounded()) {
+									List<CMElementDeclaration> singleElementList =  new ArrayList<CMElementDeclaration>(); // Is allowed again so break, and it will be added in below.
+									CMElementDeclaration singleCMElement = currentElements.get(getElementOffsetInList(currentElements, childDom));
+									singleElementList.add(singleCMElement);
+									return singleElementList;
+								}
+								return null;
+							}
+							addDecl = false; // already exists, don't give as option.
+							break;
+						}
+					}
+					if(addDecl) {
+						validElements.addAll(currentElements);
+					}
+				}
+				return validElements;
+			}
+			
+			case XSModelGroup.COMPOSITOR_SEQUENCE: {
+				List<CMElementDeclaration> validElements = getNextValidElementInSequence(element, parentCMElement, modelGroup, prefix, forceUseOfPrefix, request, response, offset);
+				if(validElements == null) {
+					break;
+				}
+				return validElements;
+			}
+		}
+		return null;
+	}
+
+
+	
+	private void collectElementsFromWildcard(XSTerm term , Collection<CMElementDeclaration> elements, DOMElement rootElement, String prefix, ICompletionRequest request) {
+		XSWildcardDecl wc = (XSWildcardDecl) term;
+		Collection<String> prefixes = rootElement.getAllPrefixes();
+		boolean useOnlyOtherNamespaces = false; // ##other
+		for (String wildCardNamespace : wc.fNamespaceList) {
+			if(wildCardNamespace == null ) {
+				continue;
+			}
+			if(wildCardNamespace.equals("##other")) {
+				useOnlyOtherNamespaces = true;
+				continue;
+			}
+			for (String p : prefixes) {
+				if (prefix != null && p.equals(prefix) && useOnlyOtherNamespaces) {

```

Comment:
if p can be null, then you need to invert the equality check: prefix.equals(p)

Reply:
it will be fine `prefixes` will at the least be an empty list, so no null issue.